homeassistant/components/aurora_abb_powerone/aurora_device.py

```diff
@@ -0,0 +1,58 @@
+"""Top level class for AuroraABBPowerOneSolarPV inverters and sensors."""
+import logging
+
+from aurorapy.client import AuroraSerialClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+from .const import (
+    ATTR_DEVICE_NAME,
+    ATTR_FIRMWARE,
+    ATTR_MODEL,
+    ATTR_SERIAL_NUMBER,
+    DEFAULT_DEVICE_NAME,
+    DOMAIN,
+    MANUFACTURER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AuroraDevice(Entity):
+    """Representation of an Aurora ABB PowerOne device."""
+
+    def __init__(self, client: AuroraSerialClient, config_entry: ConfigEntry):
+        """Initialise the basic device."""
+        self.config_entry = config_entry
+        self._id = config_entry.entry_id
+        self.type = "device"
+        self.serialnum = config_entry.data.get(ATTR_SERIAL_NUMBER, "dummy sn")
+        self._sw_version = config_entry.data.get(ATTR_FIRMWARE, "0.0.0")
+        self._model = config_entry.data.get(ATTR_MODEL, "Model unknown")
```

Comment:
No need to store this. You're already storing the config entry. Just extract it from there when you need the info

Reply:
Now sorted.